Quick_Sort/Quick_Sort.js

```diff
@@ -1,27 +1,37 @@
-function Quick_Sort(array, low, high){
+//Description: Function to perform the Quick Sort algorithm on an array of numbers
+//Expected Output: returns the sorted array
 
-  if( high - low <= 1){
-        return;
-  }
-
-  var temp = low + 1;
+//Helper function used to swap values based on indices given
+const swap = (arr, a, b)=>{
+    let temp = arr[a];
+    arr[a] = arr[b];
+    arr[b] = temp;
+}
 
-  for(var i = low + 1; i < high; ++i ){
-    if( array[i] < array[low] ){
-      var t = array[temp];
-      array[temp] = array[i];
-      array[i] = t;
-      temp++
+//Helper function to get the Pivoting Element Index
+const pivotIndex=(arr, start = 0, end = arr.length - 1)=>{
+    let swapIndex = start;
+    let pivot = arr[start];  //considering start element is pivot for first run
+    for (let i = start + 1; i <= end; i++) {
+        if (pivot > arr[i]) {
+            swapIndex += 1;
+            swap(arr, swapIndex, i);
+      }
     }
-  }
-
-  array[temp-1] = [array[low], array[low] = array[temp-1]][0];
-
-  Quick_Sort(array, low, temp-1);
-  Quick_Sort(array, temp, high);
+    swap(arr, start, swapIndex);
+    return swapIndex;
+}
 
+//Function that uses recursive definition for quick sort implementation
+const quickSort=(arr, left = 0, right = arr.length - 1)=>{
+    if (left < right) {
+        let pivot = pivotIndex(arr, left, right); //get the index for picot elemenet
+        quickSort(arr, left, pivot - 1);  //run quickSort on all elements on left side of pivot index
+        quickSort(arr, pivot + 1, right); //run quickSort on all elements on right side of pivot index
+    }
+    return arr;
 }
 
-array = [4, 3, 6, 9, 5, 1, 8, 7, 2];
-Quick_Sort(array,0,array.length);
-console.log(array)
+//I/P and O/P Examples
```

Comment:
Move the input and output to the end of the file.

Reply:
@shreya the red is the code that chnaged, the I/P examples are already at the end of the file